core/arch/riscv/include/riscv.h

```diff
@@ -374,6 +374,19 @@ static inline __noprof void uret(void)
 	asm volatile("uret");
 }
 
+__noprof uint64_t read_time(void);
+
+static inline __noprof uint64_t barrier_read_counter_timer(void)
+{
+	mb();
```

Comment:
Maybe add
```C
	/* We want the timer value after pending operations have completed */
```
...to silence the checkpatch warning.

Reply:
Thanks! I've added the comments for those barriers.